plugins/lspkind: migrate to mkNeovimPlugin

saygo-png opened the pull request.

MattSturgeon reviewed (commented):
This is a difficult plugin to modernize, as it has several bespoke options and seems to be "set up" in different ways depending on the use case.

I think more of the hard-coded logic in nixvim can be dropped or deprecated, in favour of users directly interacting with freeform settings.

I'm fine with still having extra options that controls which method of setting up is used.

Alternatively, we could have different freeform settings options; one per setup method?

saygo-png commented:
Sorry, this diff got quite noisy and i forgot to update the test so there are two pushes.

- I removed the extra options you mentioned before, since they can just be defined in settings.
- Added an assertion to make sure the user only has the cmp integration enabled when cmp is also enabled, since thats the only case where enabling this option makes sense
- Thanks to the assertion doCmp is no longer needed and i just directly use cfg.cmp.enable
- I added another optional string so that no config gets added to cmp unless the integration is explicitly enabled

> Alternatively, we could have different freeform settings options; one person setup method?

I don't understand what you mean by this.

saygo-png commented:
The test is failing now because the "empty" attribute set doesn't enable the cmp plugin.
I don't really understand what the different attribute sets in the test files are meant to be, so i'm not sure if i should add the cmp plugin to the "empty" one or disable the lspkind cmp integration instead.

These test failures make me doubt the addition of this assertion. It will break existing configs, since cmp integration is enabled by default. Should i just remove the assert, or maybe switch cmp integration to false by default?

MattSturgeon commented on plugins/by-name/lspkind/default.nix:
```diff
@@ -2,93 +2,61 @@
-          '';
     };
+
+    plugins.lspkind.luaConfig.content = lib.optionalString (!cfg.cmp.enable) ''
```
I should've spotted this last review, but ideally this'd also be `mkIf`

```suggestion
    plugins.lspkind.luaConfig.content = lib.mkIf (!cfg.cmp.enable) ''
```

Less important here, though.

MattSturgeon reviewed (approved):
Awesome!
